interpreter/value_array.go

```diff
@@ -1975,3 +1985,52 @@ func (v *ArrayValue) SetType(staticType ArrayStaticType) {
 func (v *ArrayValue) Inlined() bool {
 	return v.array.Inlined()
 }
+
+// Array iterator
+
+type ArrayIterator struct {
```

Comment:
Moved over from the vm. Similar to above, vm looping instructions needed the iterator